general/cuda.hpp

```diff
@@ -15,15 +15,15 @@
 #include "../config/config.hpp"
 #include "error.hpp"
 
-#ifdef MFEM_USE_CUDA
+#if defined(__NVCC__) && defined(MFEM_USE_CUDA)
```

Comment:
Can you explain: in what situation is this necessary?

Reply:
We have an application with various packages that depend on MFEM. Some of those packages will use the MFEM CUDA features and the others will not. Ideally we would like to have a single build of CUDA enabled MFEM which may be used with packages that are compiled with NVCC and packages which are not compiled with NVCC. By adding the extra NVCC guard, we shield the non-nvcc builds from anything CUDA related.